cpp/CMakeLists.txt

```diff
@@ -248,9 +248,15 @@ Pass multiple labels by dividing with semicolons")
     "Build with zlib compression"
     ON)
 
+  if(CMAKE_VERSION VERSION_LESS 3.7)
+    set(ARROW_WITH_ZSTD_DEFAULT OFF)
+  else()
+    # ExternalProject_Add(SOURCE_SUBDIR) is available since CMake 3.7.
```

Comment:
Shouldn't We keep `ARROW_WITH_ZSTD=ON` by default and create a workaround for [SOURCE_SUBDIR](https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L1094) instead of turning it `OFF`?

cc @pitrou

Reply:
AFAIU the workaround would be to invoke CMake manually in a CONFIGURE_COMMAND. The problem then is to figure out the right options to pass to CMake.

Reply:
It's a bit annoying that the CMakeLists.txt in zstd is not at the top level, since CMake is so ubiquitous as a build tool these days